spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/TokenAuthenticationConverter.java

```diff
@@ -1,45 +1,71 @@
 package com.sap.cloud.security.xsuaa.token;
 
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
+import com.sap.cloud.security.xsuaa.extractor.AuthoritiesExtractor;
+import com.sap.cloud.security.xsuaa.extractor.DefaultAuthoritiesExtractor;
+import com.sap.cloud.security.xsuaa.extractor.LocalAuthoritiesExtractor;
 import org.springframework.core.convert.converter.Converter;
 import org.springframework.security.authentication.AbstractAuthenticationToken;
-import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.oauth2.jwt.Jwt;
 
 import com.sap.cloud.security.xsuaa.XsuaaServiceConfiguration;
+import org.springframework.util.Assert;
 
 /**
- * Converter for xsuaa jwt token that stores authorization data like scopes
- * inside the token.
+ * An authentication converter that removes the ugly application id prefix (e.g.
+ * my-application-demo!t1229) from the scopes in the JWT.
+ *
  */
 public class TokenAuthenticationConverter implements Converter<Jwt, AbstractAuthenticationToken> {
 
-	protected String appId;
-	protected boolean provideLocalScopesOnly;
+	private AuthoritiesExtractor authoritiesExtractor;
+	private String appId;
+
+	/**
+	 * Creates a new converter with the given {@link AuthoritiesExtractor}.
+	 * 
+	 * @param authoritiesExtractor
+	 *            - the extractor used to turn Jwt scopes into Spring Security
+	 *            authorities.
+	 */
+	public TokenAuthenticationConverter(AuthoritiesExtractor authoritiesExtractor) {
+		this.authoritiesExtractor = authoritiesExtractor;
+	}
 
+	/**
+	 * Creates a new converter with a new {@link DefaultAuthoritiesExtractor}
+	 * instance as default authorities extractor.
+	 * 
+	 * @param appId
+	 *            e.g. myXsAppname!t123
+	 */
 	public TokenAuthenticationConverter(String appId) {
+		authoritiesExtractor = new DefaultAuthoritiesExtractor();
 		this.appId = appId;
 	}
 
+	/**
+	 * Creates a new converter with a new {@link DefaultAuthoritiesExtractor}
+	 * instance as default authorities extractor.
+	 *
+	 * @param xsuaaServiceConfiguration
+	 *            the xsuaa configuration
+	 * @deprecated use
```

Comment:
what's the motivation to make it deprecated?

Reply:
appId is nowhere needed in the implementation.